### PR TITLE
New pull request template option

### PR DESCRIPTION
**Pull request by jonhealy1**
This introduces some ideas for a new pull request template. See https://github.com/stac-utils/stac-utils.github.io/pull/4 for some context.

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+---
+
+## Checklist
+- [ ] **Linting:** Code is formatted and linted (run `pre-commit run --all-files`).
+- [ ] **Tests:** Tests pass. I have included new tests for these changes where applicable.
+- [ ] **Edge Cases:** I have manually verified "unhappy paths" and edge cases beyond the basic success criteria (e.g., database connection timeouts, malformed input, strict mapping rejections).
+- [ ] **Documentation:** I have updated `README.md` and `CHANGELOG.md` to reflect any new environment variables, configuration changes, or breaking updates.
+- [ ] **Accountability:** I can explain the implementation logic for every line of code submitted.
+
```
Comment (gadomski) [inline pull_request_template.md]: Removing a bit of the specific instructions, since not all repos will have these things:

```suggestion
- [ ] **Linting:** Code is formatted and linted
- [ ] **Tests:** Tests pass. I have included new tests for these changes where applicable.
- [ ] **Edge Cases:** I have manually verified "unhappy paths" and edge cases beyond the basic success criteria (e.g., database connection timeouts, malformed input, strict mapping rejections).
- [ ] **Documentation:** I have updated `README.md` to reflect any new environment variables, configuration changes, or breaking updates.
- [ ] **Accountability:** I can explain the implementation logic for every line of code submitted.

```

**Review by gadomski — CHANGES_REQUESTED** [review]
Thanks for adding this! Just one small tweak and then it's good to go for me.

**Review by gadomski — APPROVED** [review]
